src/stockpile.clj

```diff
@@ -156,12 +156,11 @@
     [(->Stockpile top (AtomicLong. (inc max-id)))
      reduction]))
 
-(defn- queue-path [q id metadata]
-  (.resolve (qpath q)
-            (apply str id (when metadata ["-" metadata]))))
-
-(defn- entry-path [q entry]
-  (queue-path q (entry-id entry) (entry-meta entry)))
+(defn- entry-path
+  ([q entry] (entry-path q (entry-id entry) (entry-meta entry)))
+  ([q id metadata]
```

Comment:
Would you object to making this a different function such as `entry-path*` or some such, I don't like the idea that you could pass `(entry-path q id)` (forgetting the metadata) and have unintended consequences.